gcc/rust/typecheck/rust-hir-type-check-expr.h

```diff
@@ -676,14 +680,25 @@ class TypeCheckExpr : public TypeCheckBase
       return true;
     });
 
-    infered_array_elems = types[0];
-    for (size_t i = 1; i < types.size (); i++)
+    // Create an infer type and register it in TypeCheckContext
+    // TODO: This is very leaky. Need a refactor of some sort.
+    auto crateNum = Analysis::Mappings::get ()->get_current_crate ();
+    auto nodeId = UNKNOWN_NODEID;
+    auto hirId = mappings->get_next_hir_id ();
+    auto localDefId = UNKNOWN_LOCAL_DEFID;
+    auto mappings = Analysis::NodeMapping (crateNum, nodeId, hirId, localDefId);
+    infered_array_elems
+      = new TyTy::InferType (hirId, TyTy::InferType::InferTypeKind::GENERAL);
+    context->insert_type (mappings, infered_array_elems);
+
+    for (auto &type : types)
       {
-	infered_array_elems = infered_array_elems->unify (types.at (i));
+	infered_array_elems = infered_array_elems->unify (type);
       }
-
     for (auto &elem : types)
-      infered_array_elems->append_reference (elem->get_ref ());
+      {
```

Comment:
IIRC GNU style does not force braces around single statement loop/if. Is this something clang-format added?

Reply:
Nah, I added it myself. It's just personal preference. Thanks for looking at my PR :)